prowler/providers/azure/azure_provider_new.py

```diff
@@ -0,0 +1,270 @@
+        )
+
+    def print_credentials(self):
+        printed_subscriptions = []
+        for key, value in self.identity.subscriptions.items():
+            intermediate = key + " : " + value
+            printed_subscriptions.append(intermediate)
+        report = f"""
+This report is being generated using the identity below:
+
+Azure Tenant IDs: {Fore.YELLOW}[{" ".join(self.identity.tenant_ids)}]{Style.RESET_ALL} Azure Tenant Domain: {Fore.YELLOW}[{self.identity.domain}]{Style.RESET_ALL} Azure Region: {Fore.YELLOW}[{self.region_config.name}]{Style.RESET_ALL}
+Azure Subscriptions: {Fore.YELLOW}{printed_subscriptions}{Style.RESET_ALL}
+Azure Identity Type: {Fore.YELLOW}[{self.identity.identity_type}]{Style.RESET_ALL} Azure Identity ID: {Fore.YELLOW}[{self.identity.identity_id}]{Style.RESET_ALL}
+"""
+        print(report)
+
+    def setup_session(
+        self, az_cli_auth, sp_env_auth, browser_auth, managed_entity_auth, tenant_id
+    ):
+        # Browser auth creds cannot be set with DefaultAzureCredentials()
+        if not browser_auth:
+            if sp_env_auth:
+                self.__check_sp_creds_env_vars__()
+            try:
+                # Since the input vars come as True when it is wanted to be used, we need to inverse it since
+                # DefaultAzureCredential sets the auth method excluding the others
+                credentials = DefaultAzureCredential(
+                    exclude_environment_credential=not sp_env_auth,
+                    exclude_cli_credential=not az_cli_auth,
+                    exclude_managed_identity_credential=not managed_entity_auth,
+                    # Azure Auth using Visual Studio is not supported
+                    exclude_visual_studio_code_credential=True,
+                    # Azure Auth using Shared Token Cache is not supported
+                    exclude_shared_token_cache_credential=True,
+                    # Azure Auth using PowerShell is not supported
+                    exclude_powershell_credential=True,
+                    # set Authority of a Microsoft Entra endpoint
+                    authority=self.region_config.authority,
+                )
+            except Exception as error:
+                logger.critical("Failed to retrieve azure credentials")
+                logger.critical(
+                    f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
+                )
+                sys.exit(1)
+        else:
+            try:
+                credentials = InteractiveBrowserCredential(tenant_id=tenant_id)
+            except Exception as error:
+                logger.critical("Failed to retrieve azure credentials")
+                logger.critical(
+                    f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
+                )
+                sys.exit(1)
+
+        return credentials
+
+    def __check_sp_creds_env_vars__(self):
```

Comment:
Can you rename this function to something more readable? 

Also we need to decide whether to use the `__function_name__` notation or not since the rest is not using it.

Reply:
It is a private function, now renamed to be more explicit